src/Modal/index.jsx

```diff
@@ -100,7 +100,19 @@ class Modal extends React.Component {
 
   render() {
     return (
-      <div className={classNames(styles.modal, styles.show, { [styles['modal-open']]: this.state.open })} role="dialog" aria-modal aria-labelledby={this.headerId}>
+      <div
+        className={classNames(
+          styles.modal,
+          {
+            [styles['modal-open']]: this.state.open,
+            [styles['modal-backdrop']]: this.state.open,
+          },
+          this.state.open ? [styles.show] : [styles.fade],
```

Comment:
This totally works - you could consider grouping all the classnames that are gated on `this.state.open` together
```javascript
{
  [styles['modal-open']]: this.state.open,
  [styles['modal-backdrop']]: this.state.open,
  [styles.show]: this.state.open,
  [styles.fade]: !this.state.open,
},
```
I find this slightly more readable since everything is grouped together but it doesn't really matter!